src/providers/oracle/qgsoracleprovider.cpp

```diff
@@ -1955,7 +1955,14 @@ void QgsOracleProvider::appendGeomParam( const QgsGeometry &geom, QSqlQuery &qry
 {
   QOCISpatialGeometry g;
 
-  QByteArray wkb = geom.asWkb();
+  QgsGeometry convertedGeom;
+  QgsWkbTypes::Type geomWkbType = geom.wkbType();
+  if ( ( geomWkbType != QgsWkbTypes::CircularString ) &&
```

Comment:
Hmmm today it's true that I don't find why. I remember a vague problem about the conversion. But it's possible that I wrote this before fixing the problem... I remove it

Reply:
CircularString are segmentized if we convert it. After discussion, it seems to be the normal behaviour, so the "if" statement will be removed, but it will be necessary : 
- to forbid loading of CircularString feature in LinesString layer,
- to allow loading of CircularString layer

Now, filter used is the same for Linestring and CircularString [qgsoracleconn.cpp#L649](https://github.com/SebastienPeillet/QGIS/blob/fix_mixed_layer_oracle/src/providers/oracle/qgsoracleconn.cpp#L649). Futhermore sdo_elem_info is the attribute to differ LineString and CircularString, and there are no easy way to retrieve [SDO_interpretation](https://docs.oracle.com/database/121/SPATL/sdo_geometry-object-type.htm#GUID-270AE39D-7B83-46D0-9DD6-E5D99C045021) from it... The SDO_ELEM_INFO type cant' be converted... So the only way is to filter on something dirty like :

```sql
to_char(substr(GEOM.get_wkt(),0,14)) = 'CIRCULARSTRING';
```

And if I do so, it will need refactoring on some other functions.

Reply:
Maybe with something like [this](https://stackoverflow.com/questions/3971693/sql-query-to-determine-if-oracle-spatial-table-contains-curves) but I'm concerned about performance.

Or we let the way it is (with comment explaining why), we don't convert curve, and let curve and linestring live in the same layer. I'm wondering how it works with multicurve and multisurface...

Reply:
Yes, I saw it after posting my previous message, I didn't know about the *table* convert word. Anyway I will try it in a few days. Thanks

Reply:
Using filter like `to_char(substr(GEOM.get_wkt(),0,14)) = 'CIRCULARSTRING'` was unappropriated, due to some limitation on get_wkt() function ( `ORA-13035: Invalid data (arcs in geodetic data) in the SDO_GEOMETRY object`).

I changed it using [this](https://stackoverflow.com/questions/3971693/sql-query-to-determine-if-oracle-spatial-table-contains-curves) method.

Now curved geometry are loaded apart, so no need to convert it. I removed this part.